core/templates/pages/exploration-player-page/services/exploration-player-state.service.spec.ts

```diff
@@ -655,4 +655,24 @@ describe('Exploration Player State Service', () => {
     explorationPlayerStateService.init();
     expect(explorationPlayerStateService.version).toBe(1);
   });
+
+  it('should tell if logged out learner progress is tracked', () => {
+    expect(explorationPlayerStateService.isLoggedOutLearnerProgressTracked())
+      .toBeFalse();
+    explorationPlayerStateService.trackLoggedOutLearnerProgress();
+    expect(explorationPlayerStateService.isLoggedOutLearnerProgressTracked())
+      .toBeTrue();
+  });
+
+  it('should set unique progress url id correctly', () => {
+    spyOn(
+      editableExplorationBackendApiService,
+      'recordProgressAndFetchUniqueProgressIdOfLoggedOutLearner')
+      .and.returnValue(Promise.resolve({
+        unique_progress_url_id: '123456'
+      }));
+    expect(explorationPlayerStateService.getUniqueProgressUrlId()).toBeNull();
+    explorationPlayerStateService.setLastCompletedCheckpoint('abc');
+    explorationPlayerStateService.setUniqueProgressUrlId();
```

Comment:
Should we have expect statements to check if the URL ID was set?

Reply:
Yes, I missed it. Added it now. Thanks!